pkg/tbtc/maintainer/bitcoin_difficulty.go

```diff
@@ -0,0 +1,63 @@
+package maintainer
+
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+)
+
+var logger = log.Logger("keep-maintainer-bitcoin-difficulty")
+
+// BitcoinDifficultyChain is an interface that provides the ability to
+// communicate with the Bitcoin difficulty on-chain contract.
+type BitcoinDifficultyChain interface {
+	// Retarget adds a new epoch to the Bitcoin difficulty relay by providing
+	// a proof of the difficulty before and after the retarget.
+	Retarget(headers []bitcoin.BlockHeader) error
+}
```

Comment:
We could also extract it to `chain.go` and name it just `Chain`. This would be consistent with what we do in `pkg/sortition` and I believe that was our desired organization of packages interacting with chain for v2 client.

Reply:
Done in 13487d5617e34cb6ae4a17b0001baa452fd57229.
I moved the `BitcoinDifficultyChain` into `chain.go`. 
But I didn't name that structure `Chain`, because we will have more than one maintainer interface (e.g.`SpvChain`).
Is that ok?